src/main.rs

```diff
@@ -11,36 +11,32 @@
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/
 
-use anyhow::{Context, Result};
+use anyhow::Result;
 use clap::Parser;
+use config::Group;
 use measure::{perform_measurement, Api, MeasurementConfig};
 use shutdown::setup_shutdown_handler;
 use std::{
     cmp::{max, min},
-    time::Duration,
+    collections::HashSet,
 };
-use tonic::transport::channel::Endpoint;
+
 use utils::read_config;
 
 mod config;
 mod measure;
 mod providers;
 mod shutdown;
 mod subscriber;
+mod types;
 mod utils;
 
 #[derive(Parser)]
 #[clap(author, version, about)]
 struct Args {
-    /// Number of iterations to run.
-    #[clap(
-        long,
-        short,
-        display_order = 1,
-        default_value_t = 1000,
-        conflicts_with = "run_forever"
-    )]
-    iterations: u64,
+    /// Number of seconds to run.
+    #[clap(long, short, display_order = 1, default_value_t = 0)]
+    duration: u64,
```

Comment:
Duration is optional, so it would be better to mark it as such:
```suggestion
    duration: Option<u64>,
```

Reply:
Done